tests/unittest/test_breakpad_resource.py

```diff
@@ -392,6 +394,48 @@ def test_existing_uuid(self, client):
         # crash id we sent
         assert result.content.decode("utf-8") == "CrashID=bp-%s\n" % crash_id
 
+    def test_add_user_agent_to_metadata(self, request_generator):
+        expected_user_agent = "wow"
+        expected_raw_crash = {
+            "ProductName": "Firefox",
+            "ReleaseChannel": "nightly",
+            "Version": "60.0a1",
+            "submitted_timestamp": "2024-04-11T01:30:56.353528+00:00",
+            "metadata": {
+                "payload": "multipart",
+                "payload_compressed": "0",
+                "payload_size": 632,
+                "collector_notes": [],
+                "dump_checksums": {
+                    "upload_file_minidump": "e9cee71ab932fde863338d08be4de9dfe39ea049bdafb342ce659ec5450b69ae"
+                },
+                # This is the only attribute I care about
+                "user_agent": "wow",
+                "throttle_rule": "is_nightly",
+            },
+            "version": 2,
+            "uuid": "d7751439-6ac2-43e0-bfd8-e70300240411",
+        }
+        data, headers = multipart_encode(
+            {
+                "ProductName": "Firefox",
+                "Version": "60.0a1",
+                "ReleaseChannel": "nightly",
+                "upload_file_minidump": ("fakecrash.dump", io.BytesIO(b"abcd1234")),
+            }
+        )
+        headers["User-Agent"] = expected_user_agent
+        handle_crashreport = MagicMock()
+        crashmover = FakeCrashMover()
+        crashmover.handle_crashreport = handle_crashreport
+        bsp = BreakpadSubmitterResource(config=self.empty_config, crashmover=crashmover)
+        req = request_generator(
+            method="POST", path="/submit", headers=headers, body=data
+        )
+        resp = Response
+        bsp.on_post(req, resp)
+        handle_crashreport.assert_called_once_with(raw_crash=expected_raw_crash)
```

Comment:
This line will always fail, because there are attributes on `raw_crash` that will change every time the test runs (like `submitted_timestamp`). How can I check only for the attribute I care about, `user_agent`?

Reply:
I wouldn't use a magic mock here. The crash report data will be in `crashmover.crashmover_queue` and you can assert against the contents there.

To deal with values that change, there are a couple of ways you can do it depending on what you're trying to do. For timestamps that you care about, you could use freezegun to freeze time. We're using that in Antenna already.

https://pypi.org/project/freezegun/

For other values like uuids which are randomly generated or values you don't care about, you can compare those values against `unittest.mock.ANY`.

```python
from unittest.mock import ANY

assert "muppets" == ANY  # true
```

Reply:
Thanks! Upon furhter discussion, willkg determined that there were some remnants of an architectural change to `crashmover` left over in the tests and some further cleanup required. That's captured in https://bugzilla.mozilla.org/show_bug.cgi?id=1891047. This PR should be rebased off that one (#995 ) once it's merged.

Reply:
How I can update my tests once #995 merges: https://github.com/mozilla-services/antenna/pull/995/files#r1561495246

Example using `ANY` in `pytest`: https://github.com/mozilla-services/antenna/pull/995/files#r1561729288